src/key_schedule.rs

```diff
@@ -40,13 +41,17 @@ pub struct KeySchedule {
     need_derive_for_extract: bool,
     hash: &'static digest::Algorithm,
     hash_of_empty_message: [u8; digest::MAX_OUTPUT_LEN],
+    prefix: &'static [u8],
     pub current_client_traffic_secret: Vec<u8>,
     pub current_server_traffic_secret: Vec<u8>,
     pub current_exporter_secret: Vec<u8>,
 }
 
+const TLS13_LABEL_PREFIX: &[u8] = b"tls13 ";
+const QUIC_LABEL_PREFIX: &[u8] = b"quic ";
+
 impl KeySchedule {
-    pub fn new(hash: &'static digest::Algorithm) -> KeySchedule {
+    pub fn new(hash: &'static digest::Algorithm, quic: bool) -> KeySchedule {
```

Comment:
Maybe just take the prefix directly here instead of a bool?

Reply:
In fact, I'm not even clear whether it makes sense for `quic` or `prefix` to be part of the `KeySchedule` state; it seems strictly an argument for derivation. As long as all the derive callers have access to the right prefix, that might be a cleaner solution.

Reply:
I'd agree with the key schedule knowing the protocol. But I think I'd always prefer a very simple enum to a bool for this kind of thing, because bools are vague at call sites. Then we can give the enum an `Into<&'static [u8]>` implementation to get the actual prefix bytes.